clickhouse-http-client/src/main/java/com/clickhouse/client/http/HttpUrlConnectionImpl.java

```diff
@@ -101,11 +99,48 @@ private ClickHouseHttpResponse buildResponse(ClickHouseOutputStream output, Runn
                 displayName, queryId, summary, format, timeZone);
     }
 
+    private HttpURLConnection dispatchProxy(ClickHouseProxyType proxyType) throws IOException {
+        HttpURLConnection newConn = null;
+        ClickHouseConfig c = config;
+
+        switch (proxyType.toString()) {
+            case "HTTP":
+                if ( c.getProxyHostName() == "" || c.getProxyPort() == -1 ) {
+                    log.error("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort());
+                    throw new RuntimeException(String.format("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort()));
+                }
+                Proxy webProxy
+                        = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(c.getProxyHostName(), c.getProxyPort()));
+                newConn = (HttpURLConnection) new URL(url).openConnection(webProxy);
+                break;
+            case "DIRECT":
+                newConn = (HttpURLConnection) new URL(url).openConnection(Proxy.NO_PROXY);
+                break;
+            case "SOCKS":
+                if ( c.getProxyHostName() == "" || c.getProxyPort() == -1 ) {
+                    log.error("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort());
+                    throw new RuntimeException(String.format("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort()));
+                }
+                Proxy socksProxy
+                        = new Proxy(Proxy.Type.SOCKS, new InetSocketAddress(c.getProxyHostName(), c.getProxyPort()));
+                newConn = (HttpURLConnection) new URL(url).openConnection(socksProxy);
+                break;
+            default:
+                // IGNORE
+                newConn = (HttpURLConnection) new URL(url).openConnection();
+        }
+        return newConn;
+    }
     private HttpURLConnection newConnection(String url, boolean post) throws IOException {
         ClickHouseConfig c = config;
-        HttpURLConnection newConn = c.isUseNoProxy()
-                ? (HttpURLConnection) new URL(url).openConnection(Proxy.NO_PROXY)
-                : (HttpURLConnection) new URL(url).openConnection();
+        HttpURLConnection newConn = null;
+
+        if (c.isUseNoProxy()) {
+            // support older versions
+            newConn = dispatchProxy(ClickHouseProxyType.DIRECT);
+        } else {
+           newConn = dispatchProxy(c.getProxyType());
```

Comment:
If you check the build log, right before the test failure, the `proxy_type` is somehow to set to `HTTP`, which might the cause?

```
2023-05-15 00:31:42:696 +0000 [main] DEBUG AbstractClient - Connecting to: ClickHouseNode [uri=http://localhost:32832/default, options={proxy_port=32824,proxy_hostname=localhost,proxy_type=HTTP}]@1792499687
2023-05-15 00:31:42:699 +0000 [main] DEBUG AbstractClient - Connection established: com.clickhouse.client.http.HttpUrlConnectionImpl@69da0b12
2023-05-15 00:31:42:700 +0000 [main] DEBUG ClickHouseHttpClient - Query: SELECT 1 FORMAT TabSeparated
2023-05-15 00:31:42:705 +0000 [main] DEBUG ClickHouseStreamResponse - 2 bytes skipped before closing input stream
```